pkg/pillar/cmd/logmanager/logmanager.go

```diff
@@ -738,13 +744,7 @@ func handleAppLogEvent(event logEntry, appLogs *logs.AppInstanceLogBundle) bool
 	}
 
 	logDetails := &logs.LogEntry{}
-	// XXX Is this still required. rsyslogd is now configured to do the same
-	logDetails.Content = strings.Map(func(r rune) rune {
-		if r == utf8.RuneError {
-			return -1
-		}
-		return r
-	}, event.content)
+	logDetails.Content = event.content
```

Comment:
How do we know this isn't required? Have you inserted some invalid UTF-8 strings in the dmesg output or something else which feeds into rsyslogd?

Reply:
rsyslogd as mmutf8fix module enables. This module replaces any invalid utf-8 sequences with a space.

Tested by sending a valid utf8
- logger $'\xe2\x80\xa2'

Tested by sending invalid utf8
- logger $'\xe2\x80\xa2'
I saw this invalid sequence replaced with space by rsyslogd.

Reply:
OK